pkg/exporter/sumologicexporter/sender.go

```diff
@@ -168,26 +168,56 @@ func (s *sender) send(ctx context.Context, pipeline PipelineType, body io.Reader
 }
 
 func (s *sender) handleReceiverResponse(resp *http.Response) error {
-	if resp.StatusCode >= 200 && resp.StatusCode < 400 {
+	if resp.ContentLength == 0 && (resp.StatusCode == 200 || resp.StatusCode == 204) {
 		return nil
 	}
 
+	type ReceiverResponseCore struct {
+		Status  int    `json:"status,omitempty"`
+		ID      string `json:"id,omitempty"`
+		Code    string `json:"code,omitempty"`
+		Message string `json:"message,omitempty"`
+	}
+
 	switch resp.StatusCode {
+	case 200, 204:
+		var rResponse ReceiverResponseCore
+		var (
+			b  = bytes.NewBuffer(make([]byte, 0, resp.ContentLength))
+			tr = io.TeeReader(resp.Body, b)
+		)
+
+		if err := json.NewDecoder(tr).Decode(&rResponse); err != nil {
+			s.logger.Warn("Error decoding receiver response", zap.ByteString("body", b.Bytes()))
+			return nil
+		}
+
+		l := s.logger.With(zap.String("status", resp.Status))
+		if len(rResponse.ID) > 0 {
+			l = l.With(zap.String("id", rResponse.ID))
+		}
+		if len(rResponse.Code) > 0 {
+			l = l.With(zap.String("code", rResponse.Code))
+		}
+		if len(rResponse.Message) > 0 {
+			l = l.With(zap.String("message", rResponse.Message))
+		}
+		l.Warn("There was an issue sending data")
```

Comment:
Aren't we going to always log this, even if we successfully sent the data?

Reply:
Nope. The contract with the API is that we get a content length of 0 and a 200 or 204 status code if the data was received successfully.

If something was off then the body will contain a json with a message.

Reply:
Can you add a comment explaining this?

Reply:
Please check now